lux/lib/lux/prisms/discord/messages/bulk_delete_messages_prism.ex

```diff
@@ -0,0 +1,155 @@
+      type: :object,
+      properties: %{
+        channel_id: %{
+          type: :string,
+          description: "The ID of the channel containing the messages",
+          pattern: "^[0-9]{17,20}$"
+        },
+        message_ids: %{
+          type: :array,
+          description: "List of message IDs to delete (2-100 messages, not older than 2 weeks)",
+          items: %{
+            type: :string,
+            pattern: "^[0-9]{17,20}$"
+          },
+          minItems: 2,
+          maxItems: 100
+        },
+        reason: %{
+          type: :string,
+          description: "The reason for deleting the messages (appears in audit log)",
+          maxLength: 512
+        }
+      },
+      required: ["channel_id", "message_ids"]
+    },
+    output_schema: %{
+      type: :object,
+      properties: %{
+        deleted: %{
+          type: :boolean,
+          description: "Whether the messages were successfully deleted"
+        },
+        count: %{
+          type: :integer,
+          description: "Number of messages deleted"
+        }
+      },
+      required: ["deleted", "count"]
+    }
+
+  alias Lux.Lenses.DiscordLens
+  require Logger
+
+  # Discord API error codes
+  @discord_errors %{
+    10003 => "Unknown channel",
+    10008 => "Unknown message",
+    50001 => "Missing access",
+    50013 => "Missing permissions",
+    50034 => "Messages provided are too old for bulk deletion",
+    50035 => "Invalid form body",
+    50021 => "Cannot execute action on this channel type"
+  }
+
+  @doc """
+  Handles the request to bulk delete messages in a Discord channel.
+  """
+  def handler(%{channel_id: channel_id, message_ids: message_ids} = input, %{agent: agent} = _ctx) do
+    Logger.info("Agent #{agent.name} bulk deleting #{length(message_ids)} messages in channel #{channel_id}")
+
+    headers = if input[:reason], do: [{"X-Audit-Log-Reason", input.reason}], else: []
+
+    case DiscordLens.focus(%{
+      endpoint: "/channels/#{channel_id}/messages/bulk-delete",
+      method: :post,
+      body: %{messages: message_ids},
+      headers: headers
+    }) do
+      {:ok, _} ->
+        count = length(message_ids)
+        Logger.info("Successfully deleted #{count} messages in channel #{channel_id}")
+        {:ok, %{
+          deleted: true,
+          count: count
+        }}
+
+      {:error, reason} ->
+        Logger.error("Failed to bulk delete Discord messages: #{inspect(reason)}")
+        handle_discord_error(reason)
+    end
+  end
+
+  @doc """
+  Validates the input parameters.
+  """
+  def validate(input) do
+    if not (Map.has_key?(input, :channel_id) and Map.has_key?(input, :message_ids)) do
+      {:error, "Missing required fields: channel_id, message_ids"}
+    else
+      with {:ok, _} <- validate_channel_id(input.channel_id),
+           {:ok, _} <- validate_message_ids(input.message_ids) do
+        :ok
+      end
+    end
+  end
+
+  defp validate_channel_id(channel_id) when is_binary(channel_id) do
```

Comment:
It would be best if we could unit test all these functions. By writing a discord client to be reused, and testing it thoroughly, you make sure that the rest of the work will go smoothly.